.github/workflows/rust.yml

```diff
@@ -374,9 +374,7 @@ jobs:
           MIRIFLAGS: "-Zmiri-disable-isolation"
         run: |
           cargo miri setup
-          cargo clean
-          # Ignore MIRI errors until we can get a clean run
-          cargo miri test || true
```

Comment:
The MIRI test on master fails to even start. For example: https://github.com/apache/arrow-datafusion/runs/7932578950?check_suite_focus=true

```
A libstd for Miri is now available in `/home/runner/.cache/miri/HOST`.
    Updating crates.io index
error: failed to select a version for the requirement `parquet = "^20.0.0"`
candidate versions found which didn't match: 15.0.0, 14.0.0, 13.0.0, ...
location searched: crates.io index
required by package `datafusion-common v11.0.0 (/home/runner/work/arrow-datafusion/arrow-datafusion/datafusion/common)`
```